lib/core/utils/select.js

```diff
@@ -103,33 +100,26 @@ function reduceIncludes(includes) {
 axe.utils.select = function select(selector, context) {
 	'use strict';
 
-	var result = [],
-		candidate;
+	let result = [];
+	let candidate;
 	if (axe._selectCache) {
 		// if used outside of run, it will still work
-		for (var j = 0, l = axe._selectCache.length; j < l; j++) {
+		for (let j = 0, l = axe._selectCache.length; j < l; j++) {
 			// First see whether the item exists in the cache
-			let item = axe._selectCache[j];
+			const item = axe._selectCache[j];
 			if (item.selector === selector) {
 				return item.result;
 			}
 		}
 	}
-	var curried = (function(context) {
+	const curried = (function(context) {
 		return function(node) {
 			return isNodeInContext(node, context);
 		};
 	})(context);
-	var reducedIncludes = reduceIncludes(context.include);
-	for (var i = 0; i < reducedIncludes.length; i++) {
+	const reducedIncludes = reduceIncludes(context.include);
+	for (let i = 0; i < reducedIncludes.length; i++) {
 		candidate = reducedIncludes[i];
-		if (
-			candidate.actualNode.nodeType === candidate.actualNode.ELEMENT_NODE &&
-			axe.utils.matchesSelector(candidate.actualNode, selector) &&
-			curried(candidate)
-		) {
-			result = pushNode(result, [candidate]);
```

Comment:
Can you explain why this is removed?

Reply:
I needed to convert it to `querySelectorAll` for virtual node only runs, but from all my testing `querySelectorAll` already returns the top node if it matches the selector, so I'm not even sure what this code did that the next line didn't do. 

The only use case I could come up with is if the selector included an element that wasn't in the axe tree but was in the DOM, but then the next line would not return any other nodes. There wasn't any test or comment about why it was there so I just removed it.